teps/0104-propagating-parameters.md

```diff
@@ -0,0 +1,638 @@
+        taskSpec:
+          params:
+          - name: HELLO
+            type: string
+            default: "Sasa World!"
+          steps:
+            - name: echo
+              image: ubuntu
+              script: |
+                #!/usr/bin/env bash
+                echo "$(params.HELLO)"          # → "Sasa World!"
+                echo "$(pipeline.params.HELLO)" # → "Hello World!"
+      - name: echo-bye
+        taskSpec:
+          steps:
+            - name: echo
+              image: ubuntu
+              script: |
+                #!/usr/bin/env bash
+                echo "$(pipeline.params.BYE)"   # → "Bye World!"
+```
+
+### Referenced Resources
+
+Implicit `Parameters` initially mutated reusable authoring-time resources that are referenced at runtime. We 
+[decided][pr-4484] to remove support for implicit `Parameters` in referenced specifications, and only support it in 
+embedded specifications. This is primarily because the behavior becomes opaque when users can't see the 
+relationship between `Parameters` declared in the referenced resources and the `Parameters` supplied in runtime
+resources. Therefore, the propagation of `Parameters` will only work for embedded or inline specifications. When a
+`PipelineRun` definition has referenced specifications but does not explicitly pass `Parameters`, the `PipelineRun`
+will be created but the execution will fail because of missing `Parameters`.
+
+```yaml
+# Invalid PipelineRun attempting to propagate Parameters to referenced Tasks
+apiVersion: tekton.dev/v1beta1
+kind: PipelineRun
+metadata:
+  generateName: pr-echo-
+spec:
+  params:
+  - name: HELLO
+    value: "Hello World!"
+  - name: BYE
+    value: "Bye World!"
+  pipelineSpec:
+    tasks:
+      - name: echo-hello
+        taskRef:
+          name: echo-hello
+      - name: echo-bye
+        taskRef:
+          name: echo-bye
+---
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  name: echo-hello
+spec:
+  steps:
+    - name: echo
+      image: ubuntu
+      script: |
+        #!/usr/bin/env bash
+        echo "$(pipeline.params.HELLO)"
+---
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  name: echo-bye
+spec:
+  steps:
+    - name: echo
+      image: ubuntu
+      script: |
+        #!/usr/bin/env bash
+        echo "$(pipeline.params.BYE)"     
+---
+# Failed execution of the above PipelineRun
+apiVersion: tekton.dev/v1beta1
+kind: PipelineRun
+metadata:
+  name: pr-echo-24lmf
+  ...
+spec:
+  params:
+  - name: HELLO
+    value: Hello World!
+  - name: BYE
+    value: Bye World!
+  pipelineSpec:
+    tasks:
+    - name: echo-hello
+      taskRef:
+        kind: Task
+        name: echo-hello
+    - name: echo-bye
+      taskRef:
+        kind: Task
+        name: echo-bye
+status:
+  conditions:
+  - lastTransitionTime: "2022-04-07T20:24:51Z"
+    message: 'invalid input params for task echo-hello: missing values for
+              these params which have no default values: [HELLO]'
```

Comment:
The message seems weird / out-of-sync compared to what is described 🤔 
Shouldn't it error out before, because we are using `$(pipelines.params)` in a `Task` definition ?

Reply:
that was left over while changing the design to use the `pipeline.params` variable instead of `params` - it's resolved when reverting to `params` - thank you!